### PR TITLE
41 feature produce first fine tuned checkpoint and validation summary for 3 5 action classes

### LINKED ISSUE #41: [Feature] Produce first fine-tuned checkpoint and validation summary for 3-5 action classes
### Goal

Produce the first usable checkpoint for Sprint 2 and document a minimal validation summary for the selected action classes.

### Scope

In scope:
- run at least one fine-tuning experiment on the chosen subset
- export the best checkpoint
- save a short validation summary with core metrics
- document class list actually used in Sprint 2

### Definition of Done

- [ ] Best checkpoint is exported to a documented path
- [ ] Validation summary is saved in docs or logs
- [ ] Selected 3-5 classes are explicitly documented
- [ ] The checkpoint can be loaded for later inference
- [ ] Tested locally
- [ ] Docs/README updated if relevant
- [ ] PR created and linked to this issue

### PR DESCRIPTION
**Pull request by XEN00000**
# [Feature] Sprint 2 Validation Summary and Checkpoint Export

## Summary
This PR fulfills the requirement to produce and validate the first usable checkpoint for Sprint 2 (Issue #41). 

**Key changes:**
* **Validation Pipeline:** Added `scripts/validate.py` to evaluate `.pth` checkpoints. The script now includes full docstrings, PEP 8 compliant imports, and proper exit codes for CI/CD compatibility.
* **Results Export:** Core metrics (accuracy, class list, sample count) are exported to a consistent location: `data/logs/metrics/validation_summary.json`.
* **Dynamic Configuration:** The model now dynamically infers `num_classes` from the dataset manifest, ensuring consistency between architecture and data subsets. This behavior is documented in the baseline docs.
* **Documentation:** Created `docs/ml-baseline.md` as the source of truth for training/validation procedures, replacing outdated README placeholders.

## Training Results (Proof of Work)
As requested for the final review, the model was fine-tuned for **50 epochs** on the local RTX environment:
* **Checkpoint:** `baseline_epoch_50.pth` (Available on OneDrive PŁ - [https://tulodz-my.sharepoint.com/:u:/g/personal/251592_edu_p_lodz_pl/IQDhTFHVg5BITIM_AfizNc7aAe644z4UFUpBEnH-WszgPk4?e=eKNetd](https://tulodz-my.sharepoint.com/:u:/g/personal/251592_edu_p_lodz_pl/IQDhTFHVg5BITIM_AfizNc7aAe644z4UFUpBEnH-WszgPk4?e=eKNetd))
* **Accuracy:** **0.665** (66.5%) on 5 classes.
* **Evaluated Classes:** `car_drops_off_person`, `car_makes_u_turn`, `motorcycle_makes_u_turn`, `motorcycle_turns_right`, `person_sits_down`.
* **Sample Size:** 200 validation videos.

## Validation Command
```bash
docker compose run --rm inference python -m scripts.validate --checkpoint data/logs/checkpoints/baseline_epoch_50.pth
```
## Linked Issue
Closes #41 

## Checklist
- [x] Changes were tested locally (Verified with 50-epoch run)
- [x] CI passes (Linting and CPU tests green)
- [x] README/docs updated `(docs/ml-baseline.md` is complete)
- [x] Scope matches the linked Issue
- [x] DoD checkboxes in Issue #41 are fully synchronized

### CODE REVIEW

**Review by lukaszjecek — CHANGES_REQUESTED** [review]
1. `scripts/validate.py` is currently not runnable as written: it refers to `config/train.yml` and `config/data_pipeline.yml`, while the repository uses `configs/...` (typo), and it also uses `args.chechpoint` instead of `args.checkpoint` (typo).

2. In the README, please make the documented paths consistent and explicit: the validation command should point to the actual checkpoint location, and the output location of `validation_summary.json` should also be documented clearly.

3. The issue requires the selected 3–5 action classes to be explicitly documented. At the moment, they are only inferred dynamically in the script, but not clearly documented in README or another persistent project document.

**Review by alicjaswiers — APPROVED** [review]

**Review by lukaszjecek — APPROVED** [review]
